lib/src/test/java/io/ably/lib/test/realtime/RealtimeAuthTest.java

```diff
@@ -74,6 +75,44 @@ public void auth_client_match_tokendetails_null_clientId() {
         }
     }
 
+    /**
+     * RSA4d:  If a request by a realtime client to an authUrl results in an HTTP 403 response,
+     * or any of an authUrl request, an authCallback, or a request to Ably to exchange
+     * a TokenRequest for a TokenDetails result in an ErrorInfo with statusCode 403,
+     * as part of an attempt by the realtime client to authenticate, then the client library
+     * should transition to the FAILED state, with an ErrorInfo (with code 80019, statusCode 403,
+     * and cause set to the underlying cause) emitted with the state change and set as the connection
+     * errorReason
+     *
+     * Verify end connection state is failed
+     */
+    @Test
+    public void auth_client_fails_authorize_server_forbidden() {
```

Comment:
You should mention spec that test belongs too, like current one to `RSA4d` and `RSA4d1`, for right naming conventions, check for other tests

Reply:
You should mention it as a part of test

Reply:
Sorry but I don't follow, what would you expect to be a test name? Or what is missing?

Reply:
Check this test, and commented documentation for the test 
https://github.com/ably/ably-java/blob/73d211c1f7da1abd3e583dedbf0431625491568f/lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java#L1521-L1540 
It includes spec ID's as to where spec belongs to

Reply:
I've extended test comments and tried to kept consistent with the rest of the test class